packages/kilt-did/src/types.ts

```diff
@@ -0,0 +1,104 @@
+import type { IIdentity, SubmittableExtrinsic } from '@kiltprotocol/types'
+import { u64 } from '@polkadot/types'
+import { AnyNumber } from '@polkadot/types/types'
+import { KeyId } from './types.chain'
+
+export type KeypairType = string
+
+export interface IPublicKey {
+  publicKey: Uint8Array
+  type: KeypairType
+  // do we need the identifier?
```

Comment:
If we want to delete a public key, we need its ID, so I guess the identifier is needed here.

Reply:
I guess also to perform lookup from chain we need the ID. I don't know exactly where this `IPublicKey` would be used? With other parties? Or just internally by the DID owner?

Reply:
I checked, the id is added in the inheriting interface `KeyDetails` together with the `includedAt` block number, the rationale being that the id is not really known before submitting to chain (but can be computed with tooling I added later).